src/tap/include/tap/tapcell.h

```diff
@@ -97,85 +115,108 @@ class Tapcell
   void reset();
   int removeCells(const std::string& prefix);
 
+  void insertBoundaryCells(const BoundaryCellOptions& options);
+  void insertTapcells(const Options& options);
+
  private:
-  struct FilledSites
+  enum class EdgeType
   {
-    int yMin = 0;
-    int xMin = 0;
-    int xMax = 0;
+    Left,
+    Top,
+    Right,
+    Bottom,
+    Unknown
   };
-  // Cells placed at corners of macros & corners of core area
-  struct CornercapMasters
+  struct Edge
   {
-    odb::dbMaster* nwin_master;
-    odb::dbMaster* nwout_master;
+    EdgeType type;
+    odb::Point pt0;
+    odb::Point pt1;
   };
-  using RowFills = std::map<int, std::vector<std::vector<int>>>;
+  enum class CornerType
+  {
+    OuterBottomLeft,
+    OuterTopLeft,
+    OuterTopRight,
+    OuterBottomRight,
+    InnerBottomLeft,
+    InnerTopLeft,
+    InnerTopRight,
+    InnerBottomRight,
+    Unknown
```

Comment:
How is this used?

Reply:
This is used to annotate the edges when I'm finding the corners of the rows areas.

Reply:
I was asking about the Unknown value

Reply:
ohh, I use that before I've assigned a value, to it. I think now that the code is working, I could remove it, it was useful in debugging at the beginning.

Reply:
If you prefer, I can remove it?